node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java

```diff
@@ -119,10 +119,26 @@ private static void validateParentHosts(ApplicationId application, NodeList node
                 .collect(Collectors.toSet());
         long numNonActive = nonActiveHosts.size();
         if (numNonActive > 0) {
-            // Note: log parent hosts not ready, but do not add to exception (to avoid leaking hostnames)
-            logger.log(LogLevel.INFO, application + ": Parent hosts not ready: " + nonActiveHosts);
-            throw new ParentHostUnavailableException("Waiting for hosts to finish booting: " +
-                                                     numNonActive + "/" + parentHostnames.size() + " left.");
+            long numActive = parentHostnames.size() - numNonActive;
+            var messageBuilder = new StringBuilder()
+                    .append(numActive).append("/").append(parentHostnames.size())
+                    .append(" hosts have completed provisioning and bootstrapping, still waiting for ");
+
+            if (nonActiveHosts.size() <= 5) {
+                messageBuilder.append(nonActiveHosts.stream()
+                        .sorted()
+                        .collect(Collectors.joining(", ")));
+            } else {
+                messageBuilder.append(nonActiveHosts.stream()
+                        .sorted()
+                        .limit(3)
+                        .collect(Collectors.joining(", ")))
+                        .append(", and others");
+            }
+            var message = messageBuilder.toString();
+
+            logger.log(LogLevel.INFO, application + ": " + message);
```

Comment:
Now that the log is pretty much the same as the exception, this is not really needed, all exceptions that lead to non 200 response from deploy are logged in https://github.com/vespa-engine/vespa/blob/259fde75fd9852cf02c275447e1f4b38114d030a/configserver/src/main/java/com/yahoo/vespa/config/server/http/HttpErrorResponse.java#L36

Reply:
Application would then not be part of the log message, so I have made that part of the message (which unfortunately is redundant for the user).